src/Uno.UI/Microsoft/UI/Xaml/Controls/RatingControl/RatingControl.cs

```diff
@@ -0,0 +1,1195 @@
+					return GetNextGlyphIfNull(rifi.PointerOverPlaceholderGlyph, RatingControlStates.Placeholder);
+				case RatingControlStates.Placeholder:
+					return GetNextGlyphIfNull(rifi.PlaceholderGlyph, RatingControlStates.Set);
+				case RatingControlStates.Unset:
+					return GetNextGlyphIfNull(rifi.UnsetGlyph, RatingControlStates.Set);
+				case RatingControlStates.Null:
+					return null;
+				default:
+					return rifi.Glyph; // "Set" state
+			}
+		}
+
+		private string GetNextGlyphIfNull(string glyph, RatingControlStates fallbackType)
+		{
+			if (glyph.Length == 0)
+			{
+				if (fallbackType == RatingControlStates.Null)
+				{
+					return null;
+				}
+				return GetAppropriateGlyph(fallbackType);
+			}
+			return glyph;
+		}
+
+		private ImageSource GetAppropriateImageSource(RatingControlStates type)
+		{
+			if (!IsItemInfoPresentAndImageInfo())
+			{
+				throw new InvalidOperationException("Runtime error, tried to retrieve an image when the ItemInfo is not a RatingItemImageInfo");
+			}
+
+			RatingItemImageInfo imageInfo = ItemInfo as RatingItemImageInfo;
+
+			switch (type)
+			{
+				case RatingControlStates.Disabled:
+					return GetNextImageIfNull(imageInfo.DisabledImage, RatingControlStates.Set);
+				case RatingControlStates.PointerOverSet:
+					return GetNextImageIfNull(imageInfo.PointerOverImage, RatingControlStates.Set);
+				case RatingControlStates.PointerOverPlaceholder:
+					return GetNextImageIfNull(imageInfo.PointerOverPlaceholderImage, RatingControlStates.Placeholder);
+				case RatingControlStates.Placeholder:
+					return GetNextImageIfNull(imageInfo.PlaceholderImage, RatingControlStates.Set);
+				case RatingControlStates.Unset:
+					return GetNextImageIfNull(imageInfo.UnsetImage, RatingControlStates.Set);
+				case RatingControlStates.Null:
+					return null;
+				default:
+					return imageInfo.Image; // "Set" state
+			}
+		}
+
+		private ImageSource GetNextImageIfNull(ImageSource image, RatingControlStates fallbackType)
+		{
+			if (image == null)
+			{
+				if (fallbackType == RatingControlStates.Null)
+				{
+					return null;
+				}
+				return GetAppropriateImageSource(fallbackType);
+			}
+			return image;
+		}
+
+		private void ResetControlWidth()
+		{
+			double newWidth = CalculateTotalRatingControlWidth();
+			Control thisAsControl = this;
+			thisAsControl.Width = newWidth;
+		}
+
+		private void ChangeRatingBy(double change, bool originatedFromMouse)
+		{
+			if (change != 0.0)
+			{
+				double ratingValue = 0.0;
+				double oldRatingValue = Value;
+				if (oldRatingValue != c_noValueSetSentinel)
+				{
+					// If the Value was programmatically set to a fraction, drop that fraction before we modify it
+					if ((int)Value != Value)
+					{
+						if (change == -1)
+						{
+							ratingValue = (int)Value;
+						}
+						else
+						{
+							ratingValue = (int)Value + change;
+						}
+					}
+					else
+					{
+						oldRatingValue = ratingValue = oldRatingValue;
```

Comment:
Codacy found an issue: [Remove this useless assignment to local variable 'oldRatingValue'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6162540)